couchbase/operations/management/freeform.hxx

```diff
@@ -0,0 +1,54 @@
+/* -*- Mode: C++; tab-width: 4; c-basic-offset: 4; indent-tabs-mode: nil -*- */
+/*
+ *   Copyright 2020-2021 Couchbase, Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+#pragma once
+
+#include <couchbase/error_context/http.hxx>
+#include <couchbase/io/http_context.hxx>
+#include <couchbase/io/http_message.hxx>
+#include <couchbase/platform/uuid.h>
+#include <couchbase/service_type.hxx>
+#include <couchbase/timeout_defaults.hxx>
+
+namespace couchbase::operations::management
+{
+struct freeform_response {
+    error_context::http ctx;
+    std::map<std::string, std::string> headers{};
+    std::string body{};
+};
+
+struct freeform_request {
+    using response_type = freeform_response;
+    using encoded_request_type = io::http_request;
+    using encoded_response_type = io::http_response;
+    using error_context_type = error_context::http;
+
+    service_type type;
+    std::string method;
+    std::string path;
```

Comment:
Method could be an enum. Same applies to the whole SDK

Reply:
Yes, it could be, but on the other hand, this is freeform request, and HTTP protocol does not limit number of HTTP methods.

Reply:
That's true